crates/macros/src/build_limits.rs

```diff
@@ -0,0 +1,191 @@
+use crate::*;
+use rayon::iter::ParallelIterator;
+use std::convert::{TryFrom, TryInto};
+use syn::spanned::Spanned;
+use winrt_gen::{NamespaceTypes, TypeLimit, TypeLimits, TypeTree};
+
+pub struct BuildLimits(pub std::collections::BTreeSet<TypesDeclaration>);
+
+impl BuildLimits {
+    pub fn to_tokens_string(self) -> Result<String, proc_macro2::TokenStream> {
+        let foundation = self.0.is_empty();
```

Comment:
nit: typically predicate variables like this are in the form `is_foundation`

Reply:
Also, is this the best way to detect this case? I can't shake the feeling that handling the foundation in the way we have been is a little hacky.

Reply:
Agreed, I was really looking for a way to identify the package being compiled and only do this if `winrt` is being compiled but I couldn't find a way to identify that.